src/webots/gui/WbSimulationView.cpp

```diff
@@ -746,24 +753,27 @@ void WbSimulationView::realTime() {
   WbSimulationState::instance()->setMode(WbSimulationState::REALTIME);
 }
 
-void WbSimulationView::run() {
-  WbSimulationState::instance()->setMode(WbSimulationState::RUN);
-}
-
 void WbSimulationView::fast() {
   WbSimulationState::instance()->setMode(WbSimulationState::FAST);
 }
 
-void WbSimulationView::disableFastMode(bool disabled) {
-  WbActionManager::instance()->action(WbAction::FAST)->setEnabled(!disabled);
-  mView3D->setUserInteractionDisabled(WbAction::DISABLE_FAST_MODE, disabled);
+void WbSimulationView::disableRendering(bool disabled) {
+  WbActionManager::instance()->action(WbAction::RENDERING)->setEnabled(!disabled);
+  mView3D->setUserInteractionDisabled(WbAction::DISABLE_RENDERING, disabled);
 }
 
-void WbSimulationView::updateFastModeOverlay() {
-  if (WbSimulationState::instance()->mode() == WbSimulationState::FAST)
-    renderABlackScreen();
+void WbSimulationView::toggleRendering() {
+  if (WbSimulationState::instance()->isRendering())
+    WbSimulationState::instance()->setRendering(false);
   else
+    WbSimulationState::instance()->setRendering(true);
+}
```

Comment:
This could be simplified
```suggestion
void WbSimulationView::toggleRendering() {
  WbSimulationState::instance()->setRendering(!WbSimulationState::instance()->isRendering());
}
```